### ISSUE TITLE
Upload of qml style fails 

### ISSUE THREAD

**Issue by NyakudyaA** [kartoza/geonode#284]
# Problem

With the new functionality for changing layer style. I just tried it and it is throwing and error.

![upload error](https://user-images.githubusercontent.com/2510900/27687880-de6b62d0-5cd8-11e7-82eb-d944c5b73053.png)

Layer to update style:http://staging.geonode.kartoza.com/layers/geonode:mdg10v4
New style: 
[layer_style.zip](https://github.com/kartoza/geonode/files/1112051/layer_style.zip)

# Solution
Make sure style replacement actually work

**Comment by lucernae** [comment]
This should have been fixed by now.
@NyakudyaA could you check this again?

**Comment by NyakudyaA** [comment]
Ok will try it now

**Comment by NyakudyaA** [comment]
@lucernae  still having issues
![error](https://user-images.githubusercontent.com/2510900/27791207-3cdc3844-5ff4-11e7-8d47-92835c8ffb2f.png)

**Comment by lucernae** [comment]
Ok, I will try to change the same layer.

**Comment by lucernae** [comment]
@NyakudyaA maybe the problem is specific with this kind of layer?
Can you try reupload or replace this layer? I tried to debug it, it seems this layer have qml file registered in database, but no actual file in filesystem, I then tried to create a dummy file but it still doesn't solve the issue.

However I tried to change http://geonode.kartoza.com/layers/geonode:malawi_buildings and it works.
I need to replicate this issue on my local environment first.

**Comment by NyakudyaA** [comment]
Ok, @lucernae  I think I understand why the layer is behaving this way.  Initially, when this layer was uploaded it did not have a qml with style. How does qgis server style a layer without a style file ?.
So from the way it looks, it seems the layer is rendered with a default style and now we having a problem replacing this layer. Perhaps we could open another ticket describing this or we could make the style mandatory when a user uploads a layer?. What are your thoughts

**Comment by lucernae** [comment]
I replicated this issue and realized I made a mistake in my django view because of wrong assumptions. Django FileField in fact will always try to create a new file if given a file handle, this will create a duplicate qml files which will produce this error in the long run. Because the old qml is not managed anymore and will linger in filesystem.

I fixed this along with PR #290 , but forgot to link it to this issue.

The expected behaviour now:
- Uploading style with the layer (at first upload) is not mandatory. But if uploaded, QGIS will use that style
- If uploaded without QML style, QGIS will generate default/random style
- We can change style (already existing or not, it doesn't matter.)

In more detail (developer commentary), this is achieved by creating a new File for QML LayerFile only when it doesn't exists and set the name accordingly. If it is exists, just overwrite the QML LayerFile.

Not yet implemented, but should be there, I'm working on it:
- Display a nice message to user, or just hide the button, if a layer doesn't have existing QML styles uploaded. It's easier to just hide the 'Download style' button.
